src-control/Control/RefCount.hs

```diff
@@ -352,22 +362,23 @@ mkWeakRefFromRaw obj = WeakRef obj
 deRefWeak ::
      (RefCounted m obj, PrimMonad m)
   => HasCallStackIfDebug
-  => WeakRef obj
+  => RefCtx
```

Comment:
I was initially thinking that this arg should not need to be here since we can (in debug mode) store the RefCtx in the WeakRef.

However, on reflection I think this is the right approach even though it is asymmetric with newRef. The choice is to supply the RefCtx when creating the WeakRef or when dereferencing it. We create WeakRefs more often than we dereference them, so supplying it at the latter point should be better.